tableau/v_tableau.zendesk_tickets.sql

```diff
@@ -1,39 +1,130 @@
 USE gabby
 GO
-
 CREATE OR ALTER VIEW tableau.zendesk_tickets AS
 
+WITH original_group AS (
+  SELECT sub.ticket_id
+         ,g.[name] AS group_name
+  FROM
+      (SELECT *,
+              ROW_NUMBER() OVER(
+              PARTITION BY ticket_id, field_name
+              ORDER BY updated) AS group_rn
+              FROM gabby.zendesk.ticket_field_history fh
+              WHERE field_name = 'group_id')
+              AS sub
+  JOIN zendesk.[group] g
+  ON sub.[value] = g.id
+  WHERE sub.group_rn = 1
+ )
+
+,group_updated AS(
+  SELECT ticket_id
+        ,MAX(updated) AS group_updated
+  FROM gabby.zendesk.ticket_field_history fh
+  WHERE field_name = 'group_id'
+  GROUP BY ticket_id
+ )
+
+,original_assignee AS (
+  SELECT sub.ticket_id
+        ,sub.assignee_rn
+        ,sub.updated AS assignee_updated
+        ,u.name AS assignee_name
+        ,scw.primary_on_site_department as orig_assignee_dept
+        ,scw.primary_job AS orig_assignee_job
+        ,scw.legal_entity_name
+  FROM (
+        SELECT *,
+               ROW_NUMBER() OVER(
+               PARTITION BY ticket_id, field_name
+               ORDER BY updated) AS assignee_rn
+          FROM gabby.zendesk.ticket_field_history fh
+        WHERE field_name = 'assignee_id' )
+        AS sub
+  JOIN zendesk.[user] u
+  ON sub.[value] = u.id
+  JOIN gabby.people.staff_crosswalk_static scw
+  ON u.email = scw.userprincipalname
+  WHERE sub.assignee_rn = 1
+ )
+
+,submitter_crosswalk AS (
+  SELECT
+        a.id
+       ,c.df_employee_number
+       ,c.legal_entity_name
+       ,c.primary_job
+       ,c.primary_on_site_department
+  FROM zendesk.[user] a
+  LEFT JOIN gabby.people.staff_crosswalk c
```

Comment:
PS - I _think_ if you move all the staff JOINs to the main SELECT statement, it should perform better

Reply:
Switched to static table. Tried moving them all to the main SELECT, but it doubled the rows when I joined to staff_crosswalf_static twice. left it there commented out in case I'm missing something silly